File: backend/src/main/java/com/skewwhiffy/auraltester/notation/model/abc/ClefUpperLedgerNotes.java
package com.skewwhiffy.auraltester.notation.model.abc;

import com.skewwhiffy.auraltester.notation.model.clef.Clef;
import com.skewwhiffy.auraltester.notation.model.note.AbsoluteNote;

import java.util.List;

public record ClefUpperLedgerNotes(Clef clef) implements ClefNotes {
    @Override
    public List<AbsoluteNote> getNotes() {
        throw new RuntimeException();
    }
}

File: backend/src/main/kotlin/com/skewwhiffy/auraltester/notation/model/abc/ClefUpperLedgerNotes.kt
package com.skewwhiffy.auraltester.notation.model.abc

import com.skewwhiffy.auraltester.notation.model.clef.Clef
import com.skewwhiffy.auraltester.notation.model.note.AbsoluteNote

data class ClefUpperLedgerNotes(val clef: Clef) : ClefNotes {
    override val notes: List<AbsoluteNote>
        get() = (1..5)
            .fold(listOf(clef.highLedgerNote.downOne)) { soFar, _ -> soFar + soFar[soFar.size - 1].upOne }
            .map { it.withNoteName }
}
